docs/contributing/contributing-recipes.md

```diff
@@ -0,0 +1,113 @@
+    ```bash
+    rad recipe register myrecipe --environment myenv --resource-type Applications.Datastores/redisCaches --template-kind bicep --template-path ghcr.io/USERNAME/recipes/myrecipe:1.1.0
+    ```
+
+    **Terraform recipe via rad CLI**
+    ```bash
+    rad recipe register myrecipe --environment myenv --resource-type Applications.Datastores/redisCaches --template-kind terraform --template-path user/recipes/myrecipe --template-version "1.1.0"
+    ```
+
+    **Via Radius environment bicep**
+    ```bicep
+    import radius as radius
+    resource env 'Applications.Core/environments@2023-10-01-preview' = {
+        name: 'prod'
+        properties: {
+            compute: {
+                kind: 'kubernetes'
+                resourceId: 'self'
+                namespace: 'default'
+            }
+            recipes: {
+                'Applications.Datastores/redisCaches':{
+                    'redis-bicep': {
+                        templateKind: 'bicep'
+                        templatePath: 'https://ghcr.io/USERNAME/recipes/myrecipe:1.1.0'
+                        // Optionally set parameters for all resources calling this Recipe
+                        parameters: {
+                            port: 3000
+                        }
+                    }
+                    'redis-terraform': {
+                        templateKind: 'terraform'
+                        templatePath: 'user/recipes/myrecipe'
+                        templateVersion: '1.1.0'
+                        // Optionally set parameters for all resources calling this Recipe
+                        parameters: {
+                            port: 3000
+                        }
+                    }
+                }   
+            }
+        }
+    }
+    ```
+
+1. Use the recipe in your application and verify that it works as expected
+    ```bicep
+    resource redis 'Applications.Datastores/redisCaches@2023-10-01-preview'= {
+        name: 'myresource'
+        properties: {
+            environment: environment
+            application: application
+            recipe: {
+                name: 'myrecipe'
+             }
+        }
+    }
+    ``` 
+    
+### 3. How to add a new recipe to this repository
+
+After you have tested the recipe in an application, you can follow the steps below to add the recipe to the repository:
+
+1. Make sure there's an issue (bug or feature) raised, which sets the expectations for the contribution you are about to make.
+1. Fork the repository and create a new branch
+1. Add the recipe to the relevant directory
+    1. If the recipe is for local dev, add the recipe to the `local-dev` directory
+    1. If the recipe is for Azure, add the recipe to the `azure` directory
+    1. If the recipe is for AWS, add the recipe to the `aws` directory
+    1. If its a terraform recipe, create a directory name `<resourcetype>` under the relevant environment and add the recipe (e.g. `azure/redis/main.tf`) 
```

Comment:
I'd suggest moving the description of "repo structure" elsewhere. This is really important info, but it's going to be hard to maintain as part of the insturctions for seting a PR.

Reply:
Moved the repo structure to top level readme.md and hyperlinked here. Let me know if thats better